.github/workflows/version_file.yml

```diff
@@ -0,0 +1,36 @@
+name: HighFive Check Version File
+
+on:
+  push:
+    branches:
+      - ci_test
+      - release/**
+  pull_request:
+    branches:
+      - master
+      - release/**
+
+jobs:
+  CheckVersion:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+
+    - name: "Install libraries"
+      run: |
+        sudo apt-get -qq update
+        sudo apt-get -qq install libhdf5-dev ninja-build
+
+    - name: Build
+      run: |
+        # Will trigger `configure_file` for H5Version.hpp.
+        cmake -DHIGHFIVE_USE_BOOST=Off -B build .
+
+    - name: Test
+      run: |
+        # Check that the file hasn't changed, i.e. was updated
+        # after changing the version number.
+        ! git status | grep include/highfive/H5Version.hpp
```

Comment:
Would be nice if we could ensure that the version is also greater or equal to the latest tag.

Reply:
I think understand the motivation. It would be nice to have something that checks we didn't forget to bump the version number.

Our current workflow is to:
1. Create a commit which updates the change log and bumps the version.
2. Create a PR.
3. Merge the PR.
4. Tag the appropriate commit on `master`.

Therefore a naive solution has two issues:
* There's a brief period where the version number is strictly larger than the latest tag. Which requires as you say, that we check for greater or equal.
* Due to the greater or equal it'll initially succeed even if we forget to bump the version number. Only after we tag a commit, will it complain.

Instead it would be nicer to have a CD pipeline to create a release, it will change the version in `CMakeLists.txt` and tag consistently.

Reply:
> Instead it would be nicer to have a CD pipeline to create a release, it will change the version in CMakeLists.txt and tag consistently.

Yes, agreed. We can put this in a virtual backlog, let's get this in first.